mcs/class/Mono.Posix/Mono.Unix.Native/Syscall.cs

```diff
@@ -1228,6 +1416,357 @@ public struct EpollEvent {
 		[FieldOffset (4)]
 		public ulong u64;
 	}
+
+	[Map]
+	[StructLayout (LayoutKind.Explicit, Size=4)]
+	public struct InAddr {
+		[FieldOffset (0)] uint addr;
+
+		public InAddr (byte b0, byte b1, byte b2, byte b3)
+		{
+			addr = 0;
+			this[0] = b0;
+			this[1] = b1;
+			this[2] = b2;
+			this[3] = b3;
+		}
+		
+		public unsafe InAddr (byte[] buffer)
+		{
+			if (buffer.Length != 4)
+				throw new ArgumentException ("buffer", "buffer.Length != 4");
+			addr = 0;
+			fixed (uint* ptr = &addr)
+				Marshal.Copy (buffer, 0, (IntPtr) ptr, 4);
+		}
+
+		public unsafe byte[] ToArray ()
```

Comment:
`ToArray()` methods should be avoided, because it will require that arrays always be allocated by the GC; there's no way to reduce memory allocations or reuse an already allocated array.

This should instead be:

    public unsafe void CopyTo (byte[] destination, int startIndex, int length)